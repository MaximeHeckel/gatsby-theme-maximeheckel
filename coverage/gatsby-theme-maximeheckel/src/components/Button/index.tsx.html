<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for gatsby-theme-maximeheckel/src/components/Button/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">gatsby-theme-maximeheckel/src/components/Button</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>18/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from '@emotion/styled';
import { withTheme } from 'emotion-theming';
import { ReactNode } from 'react';
import { Theme } from '../../theme_light';
&nbsp;
interface IButtonProps {
  primary?: boolean;
  secondary?: boolean;
  theme: Theme;
  white?: boolean;
  children: ReactNode;
}
&nbsp;
const Button = styled.button`
  margin: 0;
  background-color: ${(props: IButtonProps) =&gt;
    props.primary ? props.theme.colors.blue : 'transparent'};
  color: ${(props: IButtonProps) =&gt;
    props.primary
      ? props.theme.colors.white
      : props.white
      ? <span class="branch-0 cbranch-no" title="branch not covered" >props.theme.colors.white</span>
      : props.theme.colors.blue};
  border: none;
  cursor: pointer;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  position: relative;
  padding: ${(props: IButtonProps) =&gt; (props.primary ? '0 30px' : '0 0')};
  height: 48px;
  min-width: ${(props: IButtonProps) =&gt; (props.primary ? '190px' : 'inherit')};
  font-weight: 700;
  outline: none;
  transition: ${(props: IButtonProps) =&gt;
    props.primary
      ? 'background-color 0.2s cubic-bezier(0.8, 0, 0.2, 1)'
      : '1.8s'};
  &amp;:hover {
    background-color: ${(props: IButtonProps) =&gt;
      props.primary ? props.theme.colors.black : 'transparent'};
  }
&nbsp;
  ${(props: IButtonProps) =&gt;
    props.primary
      ? ''
      : `  ::after {
    content: '';
    display: block;
    position: absolute;
    top: calc(100% + 8px);
    left: 0;
    right: 0;
    height: 2px;
    background: ${
      props.white ? <span class="branch-0 cbranch-no" title="branch not covered" >props.theme.colors.white </span>: props.theme.colors.blue
    };
    transform: scaleX(0);
    transition: transform 0.25s ease-in;
    transform-origin: right center;
    text-decoration: none;
  }
&nbsp;
  &amp;:hover::after {
    transform: scale(1);
    transform-origin: left center;
  }`}
`;
&nbsp;
export default withTheme(Button);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Aug 05 2019 23:39:07 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
