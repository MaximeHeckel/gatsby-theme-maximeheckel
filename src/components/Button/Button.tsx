import { ReactNode } from 'react';
import { motion } from 'framer-motion';
import styled from '../../utils/styled';
import React from 'react';
interface ButtonProps {
  primary?: boolean;
  secondary?: boolean;
  tertiary?: boolean;
  white?: boolean;
  danger?: boolean;
  vercel?: boolean;
  disabled?: boolean;
  children: ReactNode;
  onClick?: () => any;
}

const StyledButton = styled(motion.button)<ButtonProps>`
  height: 48px;
  margin: 0;
  border: none;
  cursor: pointer;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  position: relative;
  font-weight: 600;
  outline: none;
  padding: 0 30px;
  border-radius: 4px;

  ${(p) =>
    p.primary
      ? `
      background-color: ${p.theme.colors.blue};
      color: ${p.theme.colors.white};
      min-width: 150px;

  ${
    p.danger
      ? `
        background-color: ${p.theme.colors.red};
            `
      : ''
  }
  
  ${
    p.vercel
      ? `
          display: flex;
          align-item: center;
          color: ${p.theme.backgroundColor};
          background-color: ${p.theme.fontColor};
        `
      : ''
  }

  ${
    p.disabled
      ? `
      cursor: not-allowed;
      opacity: 0.5;
      background-color: ${p.theme.colors.gray};
      color: #000000;
      &:hover {
        transform: none;
      }
        `
      : ''
  }
  
      `
      : ''};

  ${(p) =>
    p.secondary
      ? `
      background-color: transparent;
      color: ${p.theme.colors.blue};
      border: 2px solid ${p.theme.colors.blue};
      min-width: 150px;

  ${
    p.danger
      ? `
          color: ${p.theme.colors.red};
          border-color: ${p.theme.colors.red};
        `
      : ''
  }

  ${
    p.disabled
      ? `
      cursor: not-allowed;
      opacity: 0.5;
      color: ${p.theme.colors.gray};
      border-color: ${p.theme.colors.gray};
      &:hover {
        transform: none;
      }
        `
      : ''
  }

      `
      : ''};

  ${(p) =>
    p.tertiary
      ? `
      background-color: transparent;
      color: ${p.theme.colors.blue};
      min-width: inherited;
      padding: 0px;

      ::after {
        content: '';
        display: block;
        position: absolute;
        top: 96%;
        left: 0;
        right: 0;
        height: 2px;
        background-color: ${p.theme.colors.blue};
        transform: scaleX(0);
        transition: transform 0.25s ease-in;
        transform-origin: right center;
        text-decoration: none;
      }
    
      &:hover::after {
        transform: scale(1);
        transform-origin: left center;
      }

      ${
        p.danger
          ? `
              color: ${p.theme.colors.red};
              ::after { 
                background-color: ${p.theme.colors.red};
              }
            `
          : ''
      }

      ${
        p.disabled
          ? `
              opacity: 0.5;
              cursor: not-allowed;
              color: ${p.theme.colors.gray};
              ::after { 
                background-color: ${p.theme.colors.gray};
              }
            `
          : ''
      }
          `
      : ''};
`;

const Button = React.forwardRef(
  ({ children, ...props }: ButtonProps, ref: React.Ref<HTMLButtonElement>) => {
    return (
      <StyledButton
        whileHover={props.tertiary ? {} : { scale: 0.85 }}
        transition={{ type: 'spring', duration: 0.3 }}
        ref={ref}
        {...props}
      >
        {props.vercel ? (
          <img
            style={{
              marginBottom: '0px',
              marginRight: '8px',
            }}
            alt="Vercel logo"
            src="https://assets.vercel.com/image/upload/v1588805858/repositories/vercel/logo.png"
            height="30"
          />
        ) : null}
        {children}
      </StyledButton>
    );
  }
);

Button.displayName = 'Button';

export { Button };
