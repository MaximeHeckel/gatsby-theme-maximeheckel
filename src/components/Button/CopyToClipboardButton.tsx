import { css } from '@emotion/core';
import { motion, useMotionValue, useTransform } from 'framer-motion';
import React from 'react';

export const CopyToClipboardButton = (props: { text: string }) => {
  const duration = 0.4;
  const boxVariants = {
    hover: (isChecked: boolean) => ({
      scale: 1.05,
      strokeWidth: 3,
      opacity: isChecked ? 0 : 1,
    }),
    pressed: (isChecked: boolean) => ({
      scale: 0.95,
      strokeWidth: 1,
      opacity: isChecked ? 0 : 1,
    }),
    checked: { opacity: 0 },
    unchecked: { stroke: '#949699', strokeWidth: 2, opacity: 1 },
  };

  const tickVariants = {
    pressed: (isChecked: boolean) => ({ pathLength: isChecked ? 0.85 : 0.05 }),
    checked: { pathLength: 1 },
    unchecked: { pathLength: 0 },
  };

  const [isChecked, setIsChecked] = React.useState(false);
  const pathLength = useMotionValue(0);
  const opacity = useTransform(pathLength, [0.05, 0.15], [0, 1]);

  const copyToClipboard = (content: string) => {
    const el = document.createElement(`textarea`);
    el.value = content;
    el.setAttribute(`readonly`, ``);
    el.style.position = `absolute`;
    el.style.left = `-9999px`;
    document.body.appendChild(el);
    el.select();
    document.execCommand(`copy`);
    document.body.removeChild(el);
  };

  React.useEffect(() => {
    if (isChecked) {
      setTimeout(() => setIsChecked(false), 3000);
    }
  }, [isChecked]);

  return (
    <button
      css={css`
        background: transparent;
        border: none;
        height: 25px;
        cursor: ${isChecked ? 'default' : 'pointer'};
        outline: none;
      `}
      aria-label="Copy to clipboard"
      title="Copy to clipboard"
      disabled={isChecked}
      onClick={() => {
        copyToClipboard(props.text);
        setIsChecked(true);
      }}
    >
      <motion.svg
        initial={false}
        animate={isChecked ? 'checked' : 'unchecked'}
        whileHover="hover"
        whileTap="pressed"
        transition={{ duration }}
        width="25"
        height="25"
        viewBox="0 0 25 25"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <motion.path
          d="M20.8511 9.46338H11.8511C10.7465 9.46338 9.85107 10.3588 9.85107 11.4634V20.4634C9.85107 21.5679 10.7465 22.4634 11.8511 22.4634H20.8511C21.9556 22.4634 22.8511 21.5679 22.8511 20.4634V11.4634C22.8511 10.3588 21.9556 9.46338 20.8511 9.46338Z"
          stroke="black"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          variants={boxVariants}
          custom={isChecked}
          transition={{ duration }}
        />
        <motion.path
          d="M5.85107 15.4634H4.85107C4.32064 15.4634 3.81193 15.2527 3.43686 14.8776C3.06179 14.5025 2.85107 13.9938 2.85107 13.4634V4.46338C2.85107 3.93295 3.06179 3.42424 3.43686 3.04917C3.81193 2.67409 4.32064 2.46338 4.85107 2.46338H13.8511C14.3815 2.46338 14.8902 2.67409 15.2653 3.04917C15.6404 3.42424 15.8511 3.93295 15.8511 4.46338V5.46338"
          stroke="black"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          variants={boxVariants}
          custom={isChecked}
          transition={{ duration }}
        />
        <motion.path
          d="M20 6L9 17L4 12"
          stroke="#949699"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          variants={tickVariants}
          style={{ pathLength, opacity }}
          custom={isChecked}
          transition={{ duration }}
        />
      </motion.svg>
    </button>
  );
};
