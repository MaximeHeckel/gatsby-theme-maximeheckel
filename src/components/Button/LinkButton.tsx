import { motion } from 'framer-motion';
import styled from '../../utils/styled';

const LinkButton = styled(motion.button)`
  border-radius: var(--border-radius-1);
  width: 44px;
  height: 40px;
  cursor: pointer;
  border: none;
  background-color: transparent;
  outline: none;
  transition: 0.4s;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0px;

  > * {
    margin-left: auto;
    margin-right: auto;
  }

  &:hover {
    background-color: var(--maximeheckel-colors-foreground);
  }
`;

export { LinkButton };
