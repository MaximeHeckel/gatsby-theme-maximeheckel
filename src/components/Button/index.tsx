import { withTheme } from 'emotion-theming';
import { ReactNode } from 'react';
import styled from '../../utils/styled';
interface IButtonProps {
  primary?: boolean;
  secondary?: boolean;
  white?: boolean;
  children: ReactNode;
}

const Button = styled.button<IButtonProps>`
  margin: 0;
  background-color: ${props =>
    props.primary ? props.theme.colors.blue : 'transparent'};
  color: ${props =>
    props.primary
      ? props.theme.colors.white
      : props.white
      ? props.theme.colors.white
      : props.theme.colors.blue};
  border: none;
  cursor: pointer;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  position: relative;
  padding: ${props => (props.primary ? '0 30px' : '0 0')};
  height: 48px;
  min-width: ${props => (props.primary ? '190px' : 'inherit')};
  font-weight: 700;
  outline: none;
  transition: ${props =>
    props.primary
      ? 'background-color 0.2s cubic-bezier(0.8, 0, 0.2, 1)'
      : '1.8s'};
  &:hover {
    background-color: ${props => (props.primary ? '#161617' : 'transparent')};
  }

  ${props =>
    props.primary
      ? ''
      : `  ::after {
    content: '';
    display: block;
    position: absolute;
    top: 100%;
    left: 0;
    right: 0;
    height: 2px;
    background: ${
      props.white ? props.theme.colors.white : props.theme.colors.blue
    };
    transform: scaleX(0);
    transition: transform 0.25s ease-in;
    transform-origin: right center;
    text-decoration: none;
  }

  &:hover::after {
    transform: scale(1);
    transform-origin: left center;
  }`}
`;

export default withTheme(Button);
