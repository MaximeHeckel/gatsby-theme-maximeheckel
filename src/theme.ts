const theme = {
  light: {
    backgroundColor: 'var(--maximeheckel-colors-body-0, #F6F7F8)',
    borderColor: '#f5f5f9',
    boxShadow: '0 0px 12px -6px rgba(0, 24, 40, 0.25)',
    colors: {
      blue: 'var(--maximeheckel-colors-brand, #5184f9)',
      gray: '#F8F8F9',
      pink: '#FED5D7',
      prism: {
        token: `#fff`,
        // tslint:disable-next-line:object-literal-sort-keys
        languageJavascript: `#e8696b`,
        javascript: `#e8696b`,
        background: `#292c34`,
        comment: `#5e6a76`,
        string: `#a8e2a8`,
        var: `#b3bac5`,
        number: `#e4854d`,
        constant: `#b3bac5`,
        plain: `#fff`,
        doctype: `#e8696b`,
        tag: `#e8696b`,
        keyword: `#d49fd4`,
        boolean: `#ff5874`,
        function: `#5F8DC3`,
        parameter: `#F9965D`,
        className: `#ffcf74`,
        attrName: `#bf87ba`,
        attrValue: `#a8e2a8`,
        interpolation: `#fff`,
        punctuation: `#5FA8AA`,
        ['maybe-class-name']: `#fff`,
        property: `#80cbc4`,
        propertyAccess: `#fff`,
        namespace: `#b2ccd6`,
        highlight: `rgba(255,255,255,0.07)`,
        highlightBorder: '#196FD8',
        dom: `#5F8DC3`,
        operator: `#5FA8AA`,
      },
      white: '#FFFFFF',
    },
    fontColor: 'var(--maximeheckel-colors-typeface-0, #2B2D3E)',
    foregroundColor:
      'var(--maximeheckel-colors-body-1, rgba(217, 230, 247, 0.55))',
    overlayBackground:
      'var(--maximeheckel-colors-body-2, rgba(236, 236, 236, 0.8))',
    transitionTime: 0.5,
  },

  dark: {
    backgroundColor: 'var(--maximeheckel-colors-body-0, #1b1e21)',
    borderColor: 'rgba(255, 255, 255, 0.1)',
    boxShadow: 'rgba(0,0,0,0.2) 0px 20px 40px',
    colors: {
      black: '#000000',
      blue: 'var(--maximeheckel-colors-brand, #5184f9)',
      gray: '#F8F8F9',
      pink: '#FED5D7',
      prism: {
        token: `#fff`,
        // tslint:disable-next-line:object-literal-sort-keys
        languageJavascript: `#e8696b`,
        javascript: `#e8696b`,
        background: `#0F111B`,
        comment: `#5e6a76`,
        string: `#a8e2a8`,
        var: `#b3bac5`,
        number: `#e4854d`,
        constant: `#b3bac5`,
        plain: `#fff`,
        doctype: `#e8696b`,
        tag: `#e8696b`,
        keyword: `#d49fd4`,
        boolean: `#ff5874`,
        function: `#5F8DC3`,
        parameter: `#F9965D`,
        className: `#ffcf74`,
        attrName: `#bf87ba`,
        attrValue: `#a8e2a8`,
        interpolation: `#fff`,
        punctuation: `#5FA8AA`,
        ['maybe-class-name']: `#fff`,
        property: `#80cbc4`,
        propertyAccess: `#fff`,
        namespace: `#b2ccd6`,
        highlight: `rgba(255,255,255,0.07)`,
        highlightBorder: '#3c83da',
        dom: `#5F8DC3`,
        operator: `#5FA8AA`,
      },
      white: '#FFFFFF',
    },
    fontColor: 'var(--maximeheckel-colors-typeface-0, #FFFFFF)',
    foregroundColor:
      'var(--maximeheckel-colors-body-1, rgba(14, 20, 27, 0.55))',
    overlayBackground: 'var(--maximeheckel-colors-body-2, rgba(0,0,0,0.40))',
    transitionTime: 0.5,
  },
};

export default theme;
