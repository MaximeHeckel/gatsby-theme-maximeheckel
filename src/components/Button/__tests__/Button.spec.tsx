import { cleanup, render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import 'jest-dom/extend-expect';
import React from 'react';
import Button from '../';
import theme from '../../../theme';

describe('Button', () => {
  beforeEach(cleanup);

  it('renders the primary button', () => {
    const component = render(
      <ThemeProvider theme={theme.light}>
        <Button primary={true}>Test</Button>
      </ThemeProvider>
    );
    expect(component.baseElement).toMatchSnapshot();
  });

  it('renders the secondary button', () => {
    const component = render(
      <ThemeProvider theme={theme.light}>
        <Button secondary={true}>Test</Button>
      </ThemeProvider>
    );
    expect(component.baseElement).toMatchSnapshot();
  });
});
