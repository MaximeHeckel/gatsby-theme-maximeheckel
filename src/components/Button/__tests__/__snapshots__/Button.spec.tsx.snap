// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the primary button 1`] = `
.emotion-0 {
  height: 48px;
  margin: 0;
  border: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 600;
  outline: none;
  padding: 0 30px;
  border-radius: 4px;
  background-color: var(--maximeheckel-colors-brand,#5184f9);
  color: #FFFFFF;
  min-width: 150px;
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
}

<body>
  <div>
    <button
      class="emotion-0"
    >
      Test
    </button>
  </div>
</body>
`;

exports[`Button renders the secondary button 1`] = `
.emotion-0 {
  height: 48px;
  margin: 0;
  border: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 600;
  outline: none;
  padding: 0 30px;
  border-radius: 4px;
  background-color: transparent;
  color: var(--maximeheckel-colors-brand,#5184f9);
  border: 2px solid var(--maximeheckel-colors-brand,#5184f9);
  min-width: 150px;
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
}

<body>
  <div>
    <button
      class="emotion-0"
    >
      Test
    </button>
  </div>
</body>
`;

exports[`Button renders the tertiary button 1`] = `
.emotion-0 {
  height: 48px;
  margin: 0;
  border: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  font-weight: 600;
  outline: none;
  padding: 0 30px;
  border-radius: 4px;
  background-color: transparent;
  color: var(--maximeheckel-colors-brand,#5184f9);
  min-width: inherited;
  padding: 0px;
}

.emotion-0::after {
  content: '';
  display: block;
  position: absolute;
  top: 96%;
  left: 0;
  right: 0;
  height: 2px;
  background-color: var(--maximeheckel-colors-brand,#5184f9);
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover::after {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
}

<body>
  <div>
    <button
      class="emotion-0"
    >
      Test
    </button>
  </div>
</body>
`;
